app/api/v1/endpoints/strategies.py

```
@@ -891,6 +891,44 @@ def validate_expected_return_range_update(self):
+        """Validate configuration parameters against allowed keys."""
+        allowed_keys = {
+            'risk_level', 'position_size', 'stop_loss', 'take_profit',
+            'max_positions', 'rebalance_frequency', 'allocation_percentage',
+            'volatility_threshold', 'correlation_limit', 'drawdown_limit',
+            'profit_target', 'timeframe', 'indicators', 'entry_conditions',
+            'exit_conditions', 'custom_settings'
+        }
+        
+        invalid_keys = set(v.keys()) - allowed_keys
+        if invalid_keys:
+            raise ValueError(f"Invalid configuration keys: {', '.join(invalid_keys)}")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)